ddtrace/contrib/tornado/stack_context.py

```diff
@@ -52,27 +55,45 @@ def __exit__(self, type, value, traceback):
         self.new_contexts = None
 
     def deactivate(self):
-        self.active = False
+        self._active = False
 
-    @classmethod
-    def active(cls):
+    def active(self):
         """
         Return the ``Context`` from the current execution flow. This method can be
         used inside a Tornado coroutine to retrieve and use the current tracing context.
+        If used in a separated Thread, the `_state` thread-local storage is used to
+        propagate the current Active context from the `MainThread`.
         """
-        for ctx in reversed(_state.contexts[0]):
-            if isinstance(ctx, cls) and ctx.active:
-                return ctx.context
-
-    @classmethod
-    def activate(cls, ctx):
+        io_loop = getattr(IOLoop._current, 'instance', None)
+        if io_loop is None:
```

Comment:
I don't see no related test to that case ; maybe it is already covered somewhere else?

Reply:
yes this section (and the specular for `activate(ctx)`) is tested under `tests.contrib.tornado.test_executor_decorator`. Without this section, the context propagation between threads fails and no test will pass (you'll have 2 different traces with 2 different `trace_id`)